lib/src/notes/mod.rs

```diff
@@ -1,83 +1,138 @@
-}
\ No newline at end of file
+use miden_lib::transaction::TransactionKernel;
+use miden_objects::{
+    accounts::{ Account, AccountCode, AccountId, AccountStorage, StorageSlotType },
+    assembly::{ ModuleAst, ProgramAst },
+    assets::{ Asset, AssetVault, FungibleAsset },
+    crypto::rand::{ FeltRng, RpoRandomCoin },
+    notes::{
+        Note,
+        NoteAssets,
+        NoteExecutionMode,
+        NoteInputs,
+        NoteMetadata,
+        NoteRecipient,
+        NoteScript,
+        NoteTag,
+        NoteType,
+    },
+    transaction::{ TransactionArgs, InputNote },
+    Felt,
+    NoteError,
+    Word,
+    ZERO,
+};
+use miden_tx::TransactionExecutor;
+use miden_client::{
+    client::{
+        rpc::NodeRpcClient,
+        transactions::transaction_request::{ TransactionRequest, TransactionTemplate },
+    },
+    store::Store,
+};
+use miden_client::client::Client;
+use crate::client::AzeClient;
+use crate::executor::execute_tx_and_sync;
+use crate::constants::{ BUY_IN_AMOUNT, TRANSFER_AMOUNT };
+use std::rc::Rc;
+
+pub fn create_send_card_note<R: FeltRng, N: NodeRpcClient, S: Store>(
+    client: &mut Client<N, R, S>,
+    sender_account_id: AccountId,
+    target_account_id: AccountId,
+    assets: Vec<Asset>,
+    note_type: NoteType,
+    mut rng: RpoRandomCoin,
+    cards: [[Felt; 4]; 2]
+) -> Result<Note, NoteError> {
+    let note_script = include_str!("../../contracts/notes/game/deal.masm");
+    // TODO: hide it under feature flag debug (.with_debug_mode(true))
+    let script_ast = ProgramAst::parse(note_script).unwrap();
+    let note_script = client.compile_note_script(script_ast, vec![]).unwrap();
+
+    let card_1 = cards[0];
+    let card_2 = cards[1];
+
+    let mut inputs = [card_1.as_slice(), card_2.as_slice()].concat();
+    println!("card Inputs: {:?}", inputs);
+
+    let note_inputs = NoteInputs::new(inputs).unwrap();
+    let tag = NoteTag::from_account_id(target_account_id, NoteExecutionMode::Local)?;
+    let serial_num = rng.draw_word();
+    let aux = ZERO;
+
+    // TODO: For now hardcoding notes to be public, + Also find out what encrypted notes means
+    let metadata = NoteMetadata::new(sender_account_id, NoteType::Public, tag, aux)?;
+    let vault = NoteAssets::new(assets)?;
+    let recipient = NoteRecipient::new(serial_num, note_script, note_inputs);
+
+    Ok(Note::new(vault, metadata, recipient))
+}
+
+pub fn create_play_raise_note<R: FeltRng, N: NodeRpcClient, S: Store>(
+    client: &mut Client<N, R, S>,
+    sender_account_id: AccountId,
+    target_account_id: AccountId,
+    assets: Vec<Asset>,
+    note_type: NoteType,
+    mut rng: RpoRandomCoin
+) -> Result<Note, NoteError> {
+    let note_script = include_str!("../../contracts/notes/game/raise.masm");
+    let script_ast = ProgramAst::parse(note_script).unwrap();
+    let note_script = client.compile_note_script(script_ast, vec![]).unwrap();
+
+    let note_inputs = NoteInputs::new(vec![]).unwrap();

```

Comment:
Don't we need to pass the bet amount / raise amount ?

Reply:
As of now, the bet amount was directly set in the storage slot, since it was small blind simulation. I will remove it from there and put in note script asap.